internal/distributed/proxy/httpserver/utils.go

```diff
@@ -1118,6 +1119,25 @@ func serializeByteVectors(vectorStr string, dataType schemapb.DataType, dimensio
 	return values, nil
 }
 
+func serializeFp16OrBf16Vectors(vectorStr string, dataType schemapb.DataType, dimension int64, serializeFunc func([]float32) []byte) ([][]byte, error) {
+	// try to unmarshal as [][]float32 first to make sure `[[3, 3]]` is [][]float32 instead of [][]byte
+	fp32Values := make([][]float32, 0)
+	err := json.Unmarshal([]byte(vectorStr), &fp32Values)
+	if err == nil {
```

Comment:
Is there a chance of a false positive occurring here?
the performance here is likely to deteriorate, isn't it?